### PR TITLE
[TextFields] Add non-experimental MDCBaseTextField example

### PR DESCRIPTION
**Pull request by andrewoverton**
This PR adds a non-experimental MDCBaseTextField example.

Related to #6942.

Here are some screenshots:
Simulator Screen Shot - iPhone 7 - 2019-08-12 at 13 56 25
Simulator Screen Shot - iPhone 7 - 2019-08-12 at 13 56 23

### CODE REVIEW

File: components/TextFields/examples/MDCBaseTextFieldTypicalUseExample.m [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+- (void)viewDidLoad {
+  [super viewDidLoad];
+  self.title = kExampleTitle;
+
+  if (!self.containerScheme) {
+    self.containerScheme = [[MDCContainerScheme alloc] init];
+  }
+
+  self.view.backgroundColor = self.containerScheme.colorScheme.backgroundColor;
+  CGRect textFieldFrame = CGRectMake(15, 150, CGRectGetWidth(self.view.frame) - 30, 50);
```
Comment (codeman7) [inline components/TextFields/examples/MDCBaseTextFieldTypicalUseExample.m]: Should we set this frame in `viewWillLayoutSubviews`?

File: components/TextFields/examples/MDCBaseTextFieldTypicalUseExample.m [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+@end
+
+#pragma mark - CatalogByConvention
+
+@implementation MDCBaseTextFieldTypicalExampleViewController (CatalogByConvention)
+
++ (NSDictionary *)catalogMetadata {
+  return @{
+    @"breadcrumbs" : @[ @"Tab Field", kExampleTitle ],
```
Comment (codeman7) [inline components/TextFields/examples/MDCBaseTextFieldTypicalUseExample.m]: Tip: _"Text Field"_

File: components/TextFields/examples/MDCBaseTextFieldTypicalUseExample.m [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  self.view.backgroundColor = self.containerScheme.colorScheme.backgroundColor;
+  CGRect textFieldFrame = CGRectMake(15, 150, CGRectGetWidth(self.view.frame) - 30, 50);
+  self.textField = [[MDCBaseTextField alloc] initWithFrame:textFieldFrame];
+  self.textField.borderStyle = UITextBorderStyleRoundedRect;
+  [self.view addSubview:self.textField];
+}
+
+#pragma mark - Errata
```
Comment (codeman7) [inline components/TextFields/examples/MDCBaseTextFieldTypicalUseExample.m]: Can this be removed?

**Review by codeman7 — APPROVED** [review]

**Comment by jverkoey** [comment]
Please include a screenshot in the PR description.

**Pull request merged**